pkg/result/processors/sort_results.go

```diff
@@ -13,41 +16,67 @@ import (
 // by sorting results.Issues using processor step, and chain based
 // rules that can compare different properties of the Issues struct.
 
+const (
+	fileOrderName      = "file"
+	linterOrderName    = "linter"
+	linterSeverityName = "severity"
+)
+
 var _ Processor = (*SortResults)(nil)
 
 type SortResults struct {
-	cmp comparator
-	cfg *config.Config
+	cmps map[string][]comparator
+
+	cfg *config.Output
 }
 
 func NewSortResults(cfg *config.Config) *SortResults {
-	// For sorting we are comparing (in next order): file names, line numbers,
-	// position, and finally - giving up.
 	return &SortResults{
-		cmp: ByName{
-			next: ByLine{
-				next: ByColumn{},
-			},
+		cmps: map[string][]comparator{
+			// For sorting we are comparing (in next order):
+			// file names, line numbers, position, and finally - giving up.
+			fileOrderName: {&byName{}, &byLine{}, &byColumn{}},
+			// For sorting we are comparing: linter name
+			linterOrderName: {&byLinter{}},
+			// For sorting we are comparing: severity
+			linterSeverityName: {&bySeverity{}},
 		},
-		cfg: cfg,
+		cfg: &cfg.Output,
 	}
 }
 
 // Process is performing sorting of the result issues.
 func (sr SortResults) Process(issues []result.Issue) ([]result.Issue, error) {
-	if !sr.cfg.Output.SortResults {
+	if !sr.cfg.SortResults {
 		return issues, nil
 	}
 
+	if len(sr.cfg.SortOrder) == 0 {
+		sr.cfg.SortOrder = []string{fileOrderName}
+	}
+
+	var cmps []comparator
+	for _, name := range sr.cfg.SortOrder {
+		if c, ok := sr.cmps[name]; ok {
+			cmps = append(cmps, c...)
+		}
```

Comment:
Should we hint the user if they specify a sort order that doesn't exist instead of silently accept it?

Reply:
I added a validation to handle that on the output structure.

Reply:
But I can improve that inside the `SortResult`.